services/src/util/server.rs

```diff
@@ -294,3 +300,47 @@ pub(crate) fn log_server_info() -> Result<()> {
 pub async fn not_implemented_handler() -> HttpResponse {
     HttpResponse::NotImplemented().finish()
 }
+
+pub struct SocketFd(pub RawFd);
+
+// attach the connections socket file descriptor to the connection data
+// TODO: the socket file descriptor might be re-used by a subsequent connection before we notice that
+//       the connections was closed. We maybe need a global list of all open connections and their socket file descriptors
+//       or some other way to invalidate the socket file descriptor in the old connection when a new one arrives.
+pub fn connection_init(connection: &dyn Any, data: &mut Extensions) {
+    if let Some(sock) = connection.downcast_ref::<TcpStream>() {
+        let fd = sock.as_raw_fd();
+        if let Ok(fd) = NonZeroI32::try_from(fd) {
+            let fd = RawFd::from(fd);
```

Comment:
Could you maybe propose this to the actix-web community and see if they agree that this is one way to go?

Reply:
No. As you also realized, this works only under Linux, so this is not a general solution we could propose to actix.

Reply:
Anyway, we should discuss it with them to get their experise.

Reply:
I discussed with the core decelopers. cf. the card in trello